src/Microsoft.Management.Deployment/PackageManager.idl

```diff
@@ -87,6 +87,68 @@ namespace Microsoft.Management.Deployment
         }
     }
 
+    /// State of the uninstall.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 4)]
+    enum PackageUninstallProgressState
+    {
+        /// The uninstall is queued but not yet active. Cancellation of the IAsyncOperationWithProgress in this 
+        /// state will prevent the package from downloading or installing.
+        Queued,
+        /// The uninstall is in progress. Cancellation of the IAsyncOperationWithProgress in this state will not
+        /// stop the installation or the post uninstall steps.
+        Uninstalling,
+        /// The uninstaller has completed and cleanup actions are in progress. Cancellation of the 
+        /// IAsyncOperationWithProgress in this state will not stop cleanup or roll back the uninstall.
+        PostUninstall,
+        /// The operation has completed.
+        Finished,
+    };
+
+    /// Progress object for the uninstall
+    /// DESIGN NOTE: percentage for the uninstall as a whole is purposefully not included as there is no way to 

```

Comment:
Remove copied design note as there is a progress value that would satisfy this (if there is any point at which it actually means anything).